src/renderer/qssWidget/js/qssMouseWidget.js

```diff
@@ -0,0 +1,444 @@
+                    func: "gpii.qssWidget.mouse.getIndicatorsList",
+                    args: [
+                        "{gpii.qssWidget.mouse.indicators}.model.setting"
+                    ]
+                }
+            }
+        },
+
+        events: {
+            onIndicatorClicked: null
+        },
+
+        listeners: {
+            "onIndicatorClicked.updateValue": {
+                changePath: "setting.value",
+                value: "{arguments}.0",
+                source: "fromWidget"
+            }
+        }
+    });
+
+    /**
+     * Generates the different indicators' data based on a setting .
+     * indicators are generated using the setting's `min`, `max` and `divisibleBy` properties.
+     * In case either of those is missing, no indicators will be generated.
+     * Note that items will be recomputed every time the setting changes but only items that
+     * need to be re-rendered will do so (changeApplier merges the values).
+     * @param {Object} setting - The setting for which indicators must be created
+     * @return {Object[]} - The list of data for each indicator element. In case no indicators
+     * can be generated an empty array is returned
+     */
+    gpii.qssWidget.mouse.getIndicatorsList = function (setting) {
+        if (fluid.isValue(setting)) {
+            if (!Number.isInteger(setting.schema.min) || !Number.isInteger(setting.schema.max)) {
+                return [];
+            }
+
+            var indicators = [];
+
+            for (
+                var indicatorValue = 0;
+                indicatorValue <= setting.schema.max;
+                indicatorValue = parseFloat((indicatorValue + setting.schema.divisibleBy).toPrecision(2))
+            ) {
+                if (indicatorValue === 0) {
+                    indicators.push({
+                        indicatorValue: setting.schema.min, // value cannot be 0, instead use setting minimum value
+                        isSelected: setting.schema.min === setting.value,
+                        isRecommended: setting.schema.min === setting.schema["default"]
+                    });
+                } else {
+                    indicators.push({
+                        indicatorValue: indicatorValue, // what value to be applied when selected
+                        isSelected: indicatorValue === setting.value,
+                        isRecommended: indicatorValue === setting.schema["default"]
+                    });
+                }
+            }
+            return indicators;
+        }
+    };
+
+    /**
+     * Handler for a single indicator element.
+     *
+     * Each indicator element has three states: normal, selected and default.
+     * These three states are indicated using a custom html element
+     * attribute - "data-type". Depending on the state of this attribute, different
+     * styles are applied (refer to the CSS for more info).
+     */
+    fluid.defaults("gpii.qssWidget.mouse.indicator.presenter", {
```

Comment:
This entire grade is redundant with "gpii.qssWidget.baseStepper.indicator.presenter"